src/core/namespace.c

```diff
@@ -957,9 +956,9 @@ static int mount_tmpfs(const MountEntry *m) {
         if (r < 0)
                 return r;
 
-        r = label_fix_container(entry_path, inner_path, 0);
+        r = label_fix_container(entry_path, entry_path, 0);
```

Comment:
hmm, what? not sure i follow? if you use the same path here, you don#t need label_fix_container() at all

my understanding of the previous code was that this is supposed to undo the effect of `prefix_where_needed()` basically, i.e. get the path wthout the root prefix, so that one can derive the label for the path without the prefix... 

are you saying prefix_where_needed() is a NOP?

Reply:
The problem is that this only works if all tmpfs MountEntries are set up originally using path_const. I had some changes that set up one in a helper function, so it was using path_malloc instead - and this blew up badly. Now admittedly I failed to look at prefix_where_needed so I thought this was a result of a refactor gone slightly wrong.

I think the fundamental issue here is side-stepping the accessors that we set up for MountEntry. I'll rework this a bit to avoid that.

Reply:
iow: I intend to change it so that when prefixing we save the existing value properly in a new member, checking which of the two path variables was in use, rather than always assuming it goes from _const to _malloc

Reply:
yeah, the old code is a bit dirty. 

maybe we should add label_path_const + label_path_malloc or so as third set of paths in the structure, and then have an accessor function like this:

```c
static const char *mount_entry_label_path(const MountEntry *p) {
        return p->label_path_malloc ?: 
                p->label_path_const ?: mount_entry_path(p);
}
```

And then make prefix_where_needed() save the original unprefixed target in the label_path fields when it does its thing.

that way we should always have the right path to use for labelling around

Reply:
Updated, PTAL.